File: src/0387-first-unique-character-in-a-string/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    var s_map = {};
    for (var i = 0; i < s.length; i++) {
        if (typeof(s_map[s[i]]) == 'undefined') {
            s_map[s[i]] = 0;
        }
        s_map[s[i]]++;
    }
    for (var i = 0; i < s.length; i++) {
        if (s_map[s[i]] == 1) {
            return i;
        }
    }
    return -1;
};

module.exports = firstUniqChar;

File: src/0387-first-unique-character-in-a-string/solution.go
package solution

func firstUniqChar(s string) int {
	m := make(map[string]int)
	for i := 0; i < len(s); i++ {
		m[string(s[i])]++
	}
	for i := 0; i < len(s); i++ {
		if m[string(s[i])] == 1 {
			return i
		}
	}
	return -1
}
